src/emitters/sky.cpp

```diff
@@ -0,0 +1,203 @@
+        m_turbidity = props.float_("turbidity", 3.0f);
+        m_stretch = props.float_("stretch", 1.0f);
+        m_resolution = props.int_("resolution", 512);
+        m_albedo = props.texture<Texture>("albedo", 0.2f);
+        m_sun = compute_sun_coordinates<Float>(props);
+        m_extend = props.bool_("extend", false);
+
+        if (m_turbidity < 1 || m_turbidity > 10)
+            Log(Error, "The turbidity parameter must be in the range[1,10]!");
+        if (m_stretch < 1 || m_stretch > 2)
+            Log(Error, "The stretch parameter must be in the range [1,2]!");
+        for (size_t i = 0; i < Spectrum::Size; ++i) {
+            if (m_albedo[i] < 0 || m_albedo[i] > 1)
+                Log(Error, "The albedo parameter must be in the range [0.1]");
+        }
+
+        Float sun_elevation = 0.5f * math::Pi<Float> - m_sun.elevation;
+
+        if (sun_elevation < 0)
+            Log(Error, "The sun is below the horizon -- this is not supported by the sky model.");
+
+        for (size_t i = 0; i < Spectrum::Size; i++) {
+            if constexpr (Spectrum::Size == 3)
+                m_state[i] = arhosek_rgb_skymodelstate_alloc_init(
+                    (double)sun_elevation, (double)m_turbidity, (double)m_albedo[i]);
+            else
+                m_state[i] = arhosekskymodelstate_alloc_init(
+                    (double)sun_elevation, (double)m_turbidity, (double)m_albedo[i]);
+        }
+    }
+
+    void set_scene(const Scene *scene) override {
+        m_bsphere = scene->bbox().bounding_sphere();
+        m_bsphere.radius = max(math::RayEpsilon<Float>,
+                               m_bsphere.radius * (1.f + math::RayEpsilon<Float>));
+    }
+
+    Spectrum eval(const SurfaceInteraction3f &si, Mask active) const override {
+        MTS_MASKED_FUNCTION(ProfilerPhase::EndpointEvaluate, active);
+
+        // Compute spherical coords of wi
+        SphericalCoordinates coords = from_sphere(si.wi);
+        Float theta =  math::Pi<Float> - coords.elevation / m_stretch;
+
+        if (cos(theta) <= 0) {
+            if (!m_extend)
+                return Spectrum(0.0f);
+            else
+                theta = 0.5f * math::Pi<Float> - 1e-4f;
+        }
+
+        Float cos_gamma = cos(theta) * cos(m_sun.elevation)
+            + sin(theta) * sin(m_sun.elevation)
+            + cos(coords.azimuth - m_sun.azimuth);
+
+        // Angle between the sun and the coordinates
+        Float gamma = safe_acos(cos_gamma);
+
+        Spectrum result;
+        for (size_t i = 0; i < Spectrum::Size; i++) {
+            if constexpr (Spectrum::Size == 3)
+                result[i] = (Float) (arhosek_tristim_skymodel_radiance(
+                    m_state[i], (double)theta, (double)gamma, i) / 106.856980); // (sum of Spectrum::CIE_Y)
+            else {
+                Float step_size = (MTS_WAVELENGTH_MIN - MTS_WAVELENGTH_MIN) / (Float) Spectrum::Size;
+                Float wavelength0 = MTS_WAVELENGTH_MIN + step_size * i;
```

Comment:
(When the time comes to use spectral mode), the wavelengths to evaluate are already specified in `si.wavelength`.